src/Bicep.Core/TypeSystem/Az/AzResourceTypeProvider.cs

```diff
@@ -11,6 +11,17 @@ namespace Bicep.Core.TypeSystem.Az
     public class AzResourceTypeProvider : IResourceTypeProvider
     {
         private readonly IDictionary<string, AzResourceTypeFactory?> typeFactories = new Dictionary<string, AzResourceTypeFactory?>(StringComparer.OrdinalIgnoreCase);
+        private readonly Func<string, string, IEnumerable<SerializedTypes.Concrete.TypeBase>> loadTypeFunc;
```

Comment:
I'll be adding an interface for TypeLoader to allow it to be stubbed out and avoid passing lambdas around. Since it's in another repo, I'll make that change at a later date when integrating with this codebase.